pydantic/_internal/_generate_schema.py

```diff
@@ -345,6 +345,8 @@ def _generate_schema(self, obj: Any) -> core_schema.CoreSchema:  # noqa: C901
             return self._counter_schema(obj)
         elif origin in (typing.Dict, dict):
             return self._dict_schema(obj)
+        elif is_typeddict(origin):
```

Comment:
the trick is that a _parametrized_ generic typeddict will be a _GenericAlias, which returns False from is_typeddict, but its _origin_ will return True.